TestingTools/Elements/BaseElement.cs

```diff
@@ -0,0 +1,181 @@
+    }
+
+    public void Submit()
+    {
+        waitForVisible();
+        this.FindElement(locator).Submit();
+    }
+
+    public void Click()
+    {
+        waitForVisible();
+        waitForEnabled();
+        this.FindElement(locator).Click();
+    }
+
+    public string GetAttribute(string attributeName)
+    {
+        return this.FindElement(locator).GetAttribute(attributeName);
+    }
+
+    public string GetDomAttribute(string attributeName)
+    {
+        return this.FindElement(locator).GetDomAttribute(attributeName);
+    }
+
+    public string GetDomProperty(string propertyName)
+    {
+        return this.FindElement(locator).GetDomProperty(propertyName);
+    }
+
+    public string GetCssValue(string propertyName)
+    {
+        return this.FindElement(locator).GetCssValue(propertyName);
+    }
+
+    public ISearchContext GetShadowRoot()
+    {
+        return this.FindElement(locator).GetShadowRoot();
+    }
+
+    public IWebElement FindElement(By by)
+    {
+       return Browser.Driver.GetDriver().FindElement(by);
+    }
+
+    public ReadOnlyCollection<IWebElement> FindElements(By by)
+    {
+        return Browser.Driver.GetDriver().FindElements(by);
+    }
+
+    public IWebElement GetWebElement()
+    {
+        return this.FindElement(locator);
+    }
+
+    public void ClickUsingJs()
+    {
+        waitForEnabled();
+        Browser.JsExecutor.GetJsExecutor().ExecuteScript("arguments[0].click();", this.FindElement(locator));
+    }
+
+    public ReadOnlyCollection<IWebElement> GetWebElements()
+    {
+        return this.FindElements(locator);
+    }
+
+    public void waitForEnabled()
+    {
+        var wait = new WebDriverWait(Browser.Driver.GetDriver(), new TimeSpan(0, 0, 30));
+        var element = wait.Until(condition => 
+        {
+            try {
+                var elementToBeEnabled = this.FindElement(locator);
+                return elementToBeEnabled.Enabled;
+
+            }
+            catch (StaleElementReferenceException)
+            {
+                return false;
+            }
+            catch (NoSuchElementException)
+            {
+                return false;
+            }
+            catch (ElementNotInteractableException)
+            {
+                return false;
+            }
+        });
+    }
+
+    public void waitForVisible()
+    {
+        var wait = new WebDriverWait(Browser.Driver.GetDriver(), new TimeSpan(0, 0, 30));
+        var element = wait.Until(condition => 
+        {
+            try {
+                var elementToBeVisible = this.FindElement(locator);
+                return elementToBeVisible.Displayed;
+
+            }
+            catch (StaleElementReferenceException)
+            {
+                return false;
+            }
+            catch (NoSuchElementException)
+            {
+                return false;
+            }
+        });
+    }
+
+    public void waitForNonVisible(TimeSpan waitingTime)
```

Comment:
Don't forget about method naming convention in C#: method should be started from capital letter.